extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -47,23 +65,48 @@ export const SignTransaction = () => {
     _networkPassphrase,
     _sequence,
   } = transaction;
+
   const isFeeBump = !!_innerTransaction;
-  const source = isFeeBump ? _innerTransaction._source : transaction._source;
   const memo = decodeMemo(_memo);
 
   const [isConfirming, setIsConfirming] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [currentAccount, setCurrentAccount] = useState({} as Account);
+  const [accountNotFound, setAccountNotFound] = useState(false);
+  const [isPasswordRequired, setIsPasswordRequired] = useState(false);
+
+  const accountSelectorRef = useRef<HTMLDivElement>(null);
 
   const rejectAndClose = () => {
     dispatch(rejectTransaction());
     window.close();
   };
 
   const signAndClose = async () => {
-    setIsConfirming(true);
     await dispatch(signTransaction({ transaction }));
     window.close();
   };
 
+  const verifyPasswordThenSign = async (password: string) => {
+    const confirmPasswordResp = await dispatch(confirmPassword(password));
+
+    if (confirmPassword.fulfilled.match(confirmPasswordResp)) {
+      await signAndClose();
+    }
+  };
+
+  const handleApprove = async () => {
+    setIsConfirming(true);
+
+    if (hasPrivateKey) {
+      await signAndClose();
+    } else {
+      setIsPasswordRequired(true);
```

Comment:
if we have the need private key, do what we always did. Otherwise, show VerifyAccount and get the password so we have the needed private key